File: weird_algorithm.py
def main():
    t = int(input())
    while t >= 1:
        print(f"{t}", end=" ")
        if t == 1:
            break
        t = series(t)


def series(t):
    if t % 2 == 0:
        t = t // 2
    else:
        t = (t * 3) + 1
    return t

if __name__ == '__main__':
    main()

File: weird_algorithm.cpp
#include <iostream>

long long int series(long long int t){
    if(t % 2 == 0) {
        t = t / 2;
    }
    else {
        t = (t * 3) + 1;
    }
    return t;
}

int main() {
    long long int t;
    std::cin >> t;

    while(t >= 1) {
        std::cout << t << " ";
        if (t == 1) break;
        t = series(t);
    }

    return 0;
}